File: Emulator/src/main/content/global/skill/summoning/familiar/npc/BarkerToadNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.summoning.familiar.Familiar;
import content.global.skill.summoning.familiar.FamiliarSpecial;
import core.game.node.entity.Entity;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.game.world.update.flag.context.Graphics;
import core.plugin.Initializable;

/**
 * The type Barker toad npc.
 */
@Initializable
public class BarkerToadNPC extends Familiar {

    /**
     * Instantiates a new Barker toad npc.
     */
    public BarkerToadNPC() {
        this(null, 6889);
    }

    /**
     * Instantiates a new Barker toad npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public BarkerToadNPC(Player owner, int id) {
        super(owner, id, 800, 12123, 6, WeaponInterface.STYLE_AGGRESSIVE);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new BarkerToadNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        final Entity target = special.getTarget();
        if (!canCombatSpecial(target)) {
            return false;
        }
        animate(getProperties().getAttackAnimation());
        graphics(Graphics.create(1403));
        sendFamiliarHit(target, 8, Graphics.create(1404));
        return true;
    }

    @Override
    public int[] getIds() {
        return new int[]{6889, 6890};
    }

}

File: Emulator/src/main/content/global/skill/summoning/familiar/npc/BarkerToadNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.summoning.familiar.Familiar
import content.global.skill.summoning.familiar.FamiliarSpecial
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.game.world.update.flag.context.Graphics
import core.plugin.Initializable
import shared.consts.Items
import shared.consts.NPCs

@Initializable
class BarkerToadNPC @JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.BARKER_TOAD_6889) :
    Familiar(owner, id, 800, Items.BARKER_TOAD_POUCH_12123, 6, WeaponInterface.STYLE_AGGRESSIVE) {

    override fun construct(owner: Player, id: Int): Familiar {
        return BarkerToadNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        val target = special.target
        if (!canCombatSpecial(target)) {
            return false
        }
        animate(properties.attackAnimation)
        graphics(Graphics.create(1403))
        sendFamiliarHit(target, 8, Graphics.create(1404))
        return true
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.BARKER_TOAD_6889, NPCs.BARKER_TOAD_6890)
    }
}
